### ISSUE TITLE
keep_dims vs keepdims

### ISSUE THREAD

**Issue by yaroslavvb** [tensorflow/tensorflow#6815]
There are 15 results for [keepdims](https://github.com/tensorflow/tensorflow/search?utf8=%E2%9C%93&q=keepdims) and 77 results for [keep_dims](https://github.com/tensorflow/tensorflow/search?utf8=%E2%9C%93&q=keep_dims&type=Code) in TF codebase, any chance this can be made consistent in 1.0? Ideally TF would just match numpy API and use `keepdims` @martinwicke @karpathy

**Comment by yaroslavvb** [comment]
@aselle for comment on whether it's a feasible change

**Comment by aselle** [comment]
Yes, it's possible, probably it would be handled like axis where we keep the old argument around, but make keepdims the defacto version. It's too late to remove the old arg  for 1.0, but we can definitely make both work and the document be keepdims.

**Comment by yaroslavvb** [comment]
Cool, having alias for `keepdims` would solve the user annoyance problem

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by kho** [comment]
Names like "keepdims" is a violation of PEP-8's naming convention [https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables], thus a violation of TensorFlow's style guide

> Method Names and Instance Variables
>
> Use the function naming rules: lowercase with words separated by underscores as necessary to improve readability. 

Now one has to write code in inconsistent style like

```
tf.reduce UNDERSCORE max(..., keep NO_UNDERSCORE dims=True)
```

to avoid a deprecation warning.

Given that keep_dims had been the majority in TF's code base, wouldn't also accepting keepdims (while issuing a warning to suggest "keep_dims" instead) be a better solution?

**Comment by yaroslavvb** [comment]
IMHO compatibility with numpy is more important than compatibility with PEP-8

**Comment by neilpanchal** [comment]
I am not sure why this was modified in the first place? `keep_dims` is perfectly good.

**Comment by martinwicke** [comment]
We modified it to conform to the numpy API, which for most users is much more relevant than PEP8. 

We could talk about supporting both (without deprecation warning), or even pushing the keepdims support into a tf.numpy compatibility module. I'd much rather just have one though. This bug is about that, make sure that one can use a single version throughout.

@saxenasaurabh FYI.

**Comment by apacha** [comment]
Just saying that `keepdims` - although being used in numpy - is just a really bad name and `keep_dimensions` or `keep_dims` (if you insist on non-obvious abbreviations) is much better readable (which I would give the highest priority in an API). I would rather keep the good name, and try to change numpy parameter.

**Comment by ppwwyyxx** [comment]
Btw, `tf.nn.moments` still uses `keep_dims` in latest master.

**Comment by ppwwyyxx** [comment]
Can we fix the inconsistency? `tf.nn.moments` and many other functions in `tf.nn` still uses `keep_dims`. @martinwicke
